bodhi/client/__init__.py

```diff
@@ -476,8 +493,14 @@ def download(url, **kwargs):
                     # subprocess is icky, but koji module doesn't
                     # expose this in any usable way, and we don't want
                     # to rewrite it here.
-                    args = ('koji', 'download-build', '--arch=noarch',
-                            '--arch={0}'.format(platform.machine()), build['nvr'])
+                    if all_arch == False:
+                        args = ('koji', 'download-build', '--arch=noarch',
+                                '--arch={0}'.format(platform.machine()), build['nvr'])
+                    if all_arch == True:
+                        args = ('koji', 'download-build', build['nvr'])
+                    if requested_arch is not None:
+                        args = ('koji', 'download-build', '--arch=noarch',
+                                '--arch={0}'.format(requested_arch), build['nvr'])
```

Comment:
We could reduce the number of repeated koji snippets above a bit by doing something like this:

```
args = ['koji', 'download-build']
if all_arch is False
    args += ['--arch=noarch', '--arch={0}'.format(platform.machine())]
elif requested_arch is not None:
    args += ['--arch=noarch', '--arch={0}'.format(requested_arch)]
args += [build['nvr']]
```